django/db/models/functions/comparison.py

```diff
@@ -143,6 +143,59 @@ def as_sqlite(self, compiler, connection, **extra_context):
         return super().as_sqlite(compiler, connection, function="MAX", **extra_context)
 
 
+class JSONArray(Func):
+    function = "JSON_ARRAY"
+    output_field = JSONField()
+
+    def as_sql(self, compiler, connection, **extra_context):
+        if not connection.features.has_json_object_function:
+            raise NotSupportedError(
+                "JSONArray() is not supported on this database backend."
+            )
+        return super().as_sql(compiler, connection, **extra_context)
+
+    def as_native(self, compiler, connection, *, returning, **extra_context):
+        # By default, sqlite keeps any NULL values in the array.
+        # PostgresSQL and Oracle remove NULL values from the array by default, but
+        # we can change this behavior by adding the NULL ON NULL clause.
+        # This also matches the postgres behavior of jsonb_build_array.
```

Comment:
Classic way-too-verbose comment. Let me know how to cut this down.

Reply:
IMO you can put the text in a docstring

Reply:
Done